core/trino-main/src/main/java/io/trino/execution/buffer/PageDeserializer.java

```diff
@@ -62,14 +61,15 @@ public class PageDeserializer
     public PageDeserializer(
             BlockEncodingSerde blockEncodingSerde,
             boolean compressionEnabled,
+            Decompressor decompressor,
             Optional<SecretKey> encryptionKey,
             int blockSizeInBytes)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         requireNonNull(encryptionKey, "encryptionKey is null");
         encryptionKey.ifPresent(secretKey -> checkArgument(is256BitSecretKeySpec(secretKey), "encryptionKey is expected to be an instance of SecretKeySpec containing a 256bit key"));
         input = new SerializedPageInput(
-                compressionEnabled ? Optional.of(new Lz4Decompressor()) : Optional.empty(),
+                compressionEnabled ? Optional.ofNullable(decompressor) : Optional.empty(),
```

Comment:
if compression is enabled we should expect non-null decompressor so:

```suggestion
                compressionEnabled ? Optional.of(requireNonNull(decompressor, "decompressor is null")) : Optional.empty(),
```